llmfoundry/models/hf/model_wrapper.py

```diff
@@ -76,6 +131,285 @@ def loss(self, outputs: ModelOutput, batch: Mapping):
         # loss is at index 0 in the output tuple, logits are at index 1
         return outputs[:2]
 
+    def transform_model(self, model: PreTrainedModel) -> PreTrainedModel:
+        """Transforms the model after initialization.
+
+        Args:
+            model (PreTrainedModel): The model to transform.
+
+        Returns:
+            PreTrainedModel: The transformed model.
+        """
+        return model
+
+    @classmethod
+    def build_config(
+        cls,
+        pretrained_model_name_or_path: str,
+        trust_remote_code: bool,
+        use_auth_token: bool,
+        attn_implementation: str,
+        **kwargs: Any,
+    ) -> PretrainedConfig:
+        config = AutoConfig.from_pretrained(
+            pretrained_model_name_or_path,
+            trust_remote_code=trust_remote_code,
+            use_auth_token=use_auth_token,
+            attn_implementation=attn_implementation,
+            use_cache=
+            False,  # Necessary due to https://github.com/huggingface/transformers/issues/28056
+        )
+        return config
+
+    @classmethod
+    def build_metrics(
+        cls,
+        use_train_metrics: bool,
+        additional_train_metrics: Optional[list[str]] = None,
+        additional_eval_metrics: Optional[list[str]] = None,
+    ) -> tuple[list[Metric], list[Metric]]:
+        """Builds the training and evaluation metrics for the model.
+
+        Args:
+            use_train_metrics (bool): Whether to use training metrics.
+            additional_train_metrics (Optional[List[str]]): Additional training metrics to include.
+            additional_eval_metrics (Optional[List[str]]): Additional evaluation metrics to include.
+
+        Returns:
+            Tuple[List[Metric], List[Metric]]: A tuple containing the list of training metrics and evaluation metrics.
+        """
+        from llmfoundry.utils.builders import build_metric
+
+        train_metric_names = cls.default_train_metrics + (
+            additional_train_metrics or []
+        )
+        train_metrics = [
+            build_metric(metric, {}) for metric in train_metric_names
+        ] if use_train_metrics else []
+        eval_metric_names = cls.default_eval_metrics + (
+            additional_eval_metrics or []
+        )
+        eval_metrics = [
+            build_metric(metric, {}) for metric in eval_metric_names
+        ]
+
+        return train_metrics, eval_metrics
+
+    @classmethod
+    def build_inner_model(
+        cls,
+        pretrained_model_name_or_path: str,
+        pretrained_lora_id_or_path: Optional[str],
+        trust_remote_code: bool,
+        init_device: str,
+        use_flash_attention_2: bool,
+        use_auth_token: bool,
+        config_overrides: dict[str, Any],
+        load_in_8bit: bool,
+        pretrained: bool,
```

Comment:
you removed two args here, which is backwards breaking. can we put them back?

Reply:
can disregard this since making a new class

Reply:
although might be nice to maintain it so we can just say "adjust your constructor, but otherwise the interface on the new class is the same as the interface on the causal lm class"